part-1.py

```diff
@@ -4,17 +4,66 @@
 # the appropriate comment.
 
 # factorial
-
-
+def factorial(n):
+    if n == 0:
+        return 1
+    
+    elif n > 0:
+        return n * factorial(n-1)
+    
+    else:
+        raise ValueError
 
 # reverse
-
-
+def reverse(text):
+    if len(text) <= 1:
+        return text
+    else:
+        return text[::-1]
```

Comment:
While this does work, it isn't recursively calling itself, so it's not yet recursion. We want to concatenate the string character by character with each call.